rust/src/nlp/src/npl.rs

```diff
@@ -0,0 +1,27 @@
+
+use rust_bert::pipelines::common::ModelType;
+use rust_bert::pipelines::question_answering::{QuestionAnsweringModel, QaInput};
+use std::error::Error;
+
+// process user input and extract relevant information
+pub(crate) fn process_input(input_text: &str) -> Result<String, Box<dyn Error>> {
+    //  QuestionAnsweringModel
+    let qa_model = QuestionAnsweringModel::new(ModelType::Bert);
+
+    let question = "What's the name of the bridge?";
```

Comment:
This is an interesting approach that I was not aware of, yet. Thank you for demonstrating it. Eventually, users should be able to state arbitrary commands that define the changes they want to do to the system. Therefore, the dialog probably needs to be more sophisticated to reach this goal. Example:

- Q: Does the user want to create something or change an interface?
- A: create
- Q: What type of interface does the user want to create?
- A: bridge
- Q: What's the name of the bridge...

The challenge I see here, is to have a full decision tree to cover the full capabilities. The advantage is that this will guide to a correct syntax.

What are your thoughts about this?

Reply:
@liangwen12year what's your opinion, here?

Reply:
> This is an interesting approach that I was not aware of, yet. Thank you for demonstrating it. Eventually, users should be able to state arbitrary commands that define the changes they want to do to the system. Therefore, the dialog probably needs to be more sophisticated to reach this goal. Example:
> 
>     * Q: Does the user want to create something or change an interface?
> 
>     * A: create
> 
>     * Q: What type of interface does the user want to create?
> 
>     * A: bridge
> 
>     * Q: What's the name of the bridge...
> 
> 
> The challenge I see here, is to have a full decision tree to cover the full capabilities. The advantage is that this will guide to a correct syntax.
> 
> What are your thoughts about this?

IMO, this sophistication of the dialog really would be great especially on the UX , because it would easy for them to navigate a complex configuration step by step and eventually  a accurate syntax, but as you have said the  overall config options may be relatively hard to  obtain but possibly from commonly used configs and experts  for the usecases, and best practices.

Reply:
There are several challenges behind the scenes, what if the user provides an invalid interface type and how can we do the validation on the user's input, maybe we should point out the error and ask the same question again. And the questions should be varied based on the previous dialogue context, using a decision tree approach is deterministic and can generate the profiles as accurate as possible and avoid the randomness for the result.